tests/run-make/libtest-thread-limit/rmake.rs

```diff
@@ -0,0 +1,19 @@
+// libtest used to panic if it hit the thread limit. This often resulted in spurious test failures
+// (thread 'main' panicked at 'called Result::unwrap() on an Err value: Os
+// { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }' ...
+// error: test failed, to rerun pass '--lib').
+// Since the fix in #81546, the test should continue to run synchronously
+// if it runs out of threads. Therefore, this test's final execution step
+// should succeed without an error.
+// See https://github.com/rust-lang/rust/pull/81546
+
+//@ only-linux
+// Reason: the `ulimit` utility is used
```

Comment:
Remark: technically, we would also have to check *which* shell is used if we use `ulimit -p`, because it means different things in bash vs dash, and who knows what other shells do...

Reply:
In the original PR https://github.com/rust-lang/rust/pull/81546 apparently the shell used in *one* of the CI jobs is `dash`, so `ulimit -p` is *actually* process limit here (which makes more sense than pipe size lmao)

> There's also some strange /bin/dash: 1: Cannot fork in the log. That goes away if I remove the ulimit, but I don't know why it goes away (or why it happens), and the compilation still fails... 😕
The test is fine on i586-unknown-linux-gnu (at least locally). I'm investigating this with --target i686-unknown-linux-musl now.

Reply:
Suggestion: we may have to do a linux syscall here with `setrlimit` via [`libc::setrlimit`](https://docs.rs/libc/latest/libc/fn.setrlimit.html), i.e.

```rs
pub unsafe extern "C" fn setrlimit(
    resource: __rlimit_resource_t,
    rlim: *const rlimit
) -> c_int
```

We want `resource` to be [`RLIMIT_NPROC`](https://docs.rs/libc/latest/libc/constant.RLIMIT_NPROC.html) to be the number of processes, this may have to be `0` or `1` but we need to double-check.

> `RLIMIT_NPROC`: The maximum number of processes (or, more precisely on Linux, threads) that can be created for the real user ID of the calling process. Upon encountering this limit, [fork](https://linux.die.net/man/2/fork)(2) fails with the error EAGAIN.
> -- <https://linux.die.net/man/2/setrlimit>

We want `rlimit` to either be `0` or `1` (probably `1` since it has to spawn the `test` process? not sure, needs testing):

```rs
rlimit {
    rlim_cur: 0, // or 1, needs testing
    rlim_max: 0, // or 1, needs testing
}
```

Reply:
@Mark-Simulacrum since you were the original reviewer of #81546, I just want to get a vibe check from you, does this seem right?